node/src/util.rs

```diff
@@ -83,3 +84,16 @@ pub fn get_triple_timeout() -> Duration {
         .unwrap_or_default()
         .unwrap_or(crate::types::PROTOCOL_TRIPLE_TIMEOUT)
 }
+
+pub fn is_elapsed_longer_than_timeout(timestamp_sec: u64, timeout: Duration) -> bool {
+    if let LocalResult::Single(msg_timestamp) = Utc.timestamp_opt(timestamp_sec as i64, 0) {
+        let now_datetime: DateTime<Utc> = Utc::now();
+        // Calculate the difference in seconds
+        let elapsed_duration = now_datetime.signed_duration_since(msg_timestamp);
+        let timeout = chrono::Duration::seconds(timeout.as_secs() as i64)
+            + chrono::Duration::nanoseconds(timeout.subsec_nanos() as i64);
+        elapsed_duration > timeout
+    } else {
+        false
```

Comment:
If we return false in case of failure, such message will never be deleted from the queue.

Reply:
Seems like the case where it's not `Single` only happens in the [following](https://docs.rs/chrono/latest/chrono/trait.TimeZone.html#method.timestamp_opt)
> Returns MappedLocalTime::None on out-of-range number of seconds and/or invalid nanosecond, otherwise always returns MappedLocalTime::Single.

So we should be fine for the most part since our timestamp should never be out of range since we created it with `Utc::now().timestamp()` in the first place, but maybe we should add some guardrails to the type itself later when we get the chance, just in case someone ends up setting `timestamp` to be something else not from the `chrono` crate